cmd/holo-files/internal/impl/entity_apply_nonorphan.go

```diff
@@ -36,23 +36,48 @@ import (
 //resources, and saving the result in the target path with the correct
 //file metadata.
 func (entity *Entity) applyNonOrphan(withForce bool) (skipReport bool, err error) {
-	//determine the related paths
-	var (
-		targetPath      = entity.PathIn(common.TargetDirectory())
-		basePath        = entity.PathIn(common.BaseDirectory())
-		provisionedPath = entity.PathIn(common.ProvisionedDirectory())
-	)
-
-	//step 1: entities may only be applied if:
-	//option 1: there is a manageable file in the target location (this target
-	//file is either the base from the application package or the
-	//product of a previous Apply run)
-	//option 2: the target file was deleted, but we have a base that we
-	//can start from
+	//step 1: check if a system update installed a new version of the stock
+	//configuration
+	//
+	// This has to come first because it might shuffle some files
+	// around, and if we do anything else first, we might end up
+	// stat()ing the wrong file.
+	newBasePath, newBase, err := entity.GetNewBase()
+	if err != nil {
+		return false, err
+	}
+
+	// step 2: Load our 3 versions into memory.
+	current, err := entity.GetCurrent()
+	if err != nil && !os.IsNotExist(err) {
+		if pe, ok := err.(*os.PathError); ok {
+			pe.Op, pe.Path = "skipping", "target"
+		}
```

Comment:
Only one line would get changed:

    -pe.Op, pe.Path = "skipping", "target"
    +errors.New("skipping target: " + pe.Err.Error())

(or even add more lines if using `ErrSkippingTarget`). That might be a little clearer, but I'm not sure it's better.  I'm a bit averse to turning error causes into opaque strings.

I don't think it's worth pulling in a dependency on `"github.com/pkg/errors"`, but if it wre there, I would use `err = errors.Wrap(pe.Err, "skipping target")` without a second thought.